File: Implementation/javascript/2290.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
const [s, n] = input[0].split(" ");

const answer = [];
const top = [2, 3, 5, 6, 7, 8, 9, 0];
const topColLeftRight = [4, 8, 9, 0];
const topColLeft = [5, 6];
const topColRight = [1, 2, 3, 7];

const middle = [2, 3, 4, 5, 6, 8, 9];

const bottomColLeftRight = [6, 8, 0];
const bottomColLeft = [2];
const bottomColRight = [1, 3, 4, 5, 7, 9];
const bottom = [2, 3, 5, 6, 8, 9, 0];
const row = (arr) => {
  return [...n].reduce((string, char) => {
    if (arr.includes(Number(char))) {
      return (
        string +
        " " +
        Array(Number(s))
          .fill(1)
          .map(() => "-")
          .join("") +
        " " +
        " "
      );
    }
    return (
      string +
      Array(Number(s) + 3)
        .fill(1)
        .map(() => " ")
        .join("")
    );
  }, "");
};

const col = (one, two, three) => {
  return Array(Number(s))
    .fill(1)
    .map(() => {
      return [...n].reduce((string, char) => {
        if (one.includes(Number(char))) {
          return (
            string +
            "|" +
            Array(Number(s))
              .fill(1)
              .map(() => " ")
              .join("") +
            "|" +
            " "
          );
        }
        if (two.includes(Number(char))) {
          return (
            string +
            "|" +
            Array(Number(s) + 2)
              .fill(1)
              .map(() => " ")
              .join("")
          );
        }
        if (three.includes(Number(char))) {
          return (
            string +
            Array(Number(s) + 1)
              .fill(1)
              .map(() => " ")
              .join("") +
            "|" +
            " "
          );
        }
        return (
          string +
          Array(Number(s) + 3)
            .fill(1)
            .map(() => " ")
            .join("")
        );
      }, "");
    })
    .join("\n");
};

answer.push(row(top));
answer.push(col(topColLeftRight, topColLeft, topColRight));
answer.push(row(middle));
answer.push(col(bottomColLeftRight, bottomColLeft, bottomColRight));
answer.push(row(bottom));
console.log(answer.join("\n"));

File: Implementation/python/2290.py
# 첫번째줄 : ( 2 3 5 6 7 8 9 0)
# 두번째줄 ~ S개 줄 : S+2번째에 : 1,2,3,4,7,8,9,0
# 두번째줄 ~ S개 줄 : 0번째에 : 4,5,6,8,9,0
# S+1 번째 (가로) : 2,3,4,5,6,8,9

# S+1번째 4번째 자리 : 1,3,4,5,6,7,8,9,0
# S+1번째 ~s개 줄 첫번째 자리 : 2,6,8,0
# 마지막줄 : 2,3,5,6,8,9,0


s,n = input().split()

one = [2,3,5,6,7,8,9,0]
two_three = [4,8,9,0]
two = [5,6]
three = [1,2,3,7]

four = [2,3,4,5,6,8,9]

five_six = [6,8,0]
five = [2]
six = [1,3,4,5,7,9]
seven = [2,3,5,6,8,9,0]

result = ""
def row (result,arr):
  for char in n:
    if int(char) in arr:
      result += " "
      result += "-"*int(s)
      result += " "
    else:
      result += " "*(2+int(s))
    result += " "
  result += "\n"  
  return result

def column (result,arr,arr2,arr3):
  for _ in range(int(s)):
    for char in n:
      if int(char) in arr:
        result += "|"
        result += " " * int(s)
        result += "|"
        result += " "
        continue
      if int(char) in arr2:
        result += "|"
        result += " " * int(s)
        result += " "
        result += " "
        continue
      if int(char) in arr3:
        result += " "
        result += " " * (int(s))
        result += "|"
        result += " "
        continue
      result += " " * (int(s)+1)
    result += "\n"
  return result

result = row(result,one)
result = column(result,two_three,two,three)
result = row(result,four)
result = column(result,five_six,five,six)
result = row(result,seven)
print(result)